dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -0,0 +1,239 @@
+import React, { useState, useEffect } from "react";
+import {useDispatch} from 'react-redux'
+import "./index.css";
+import {
+  ToggleGroup,
+  ToggleGroupItem,
+  Page,
+  Masthead,
+  MastheadToggle,
+  PageSidebar,
+  PageToggleButton,
+  PageSection,
+  PageSectionVariants,
+} from "@patternfly/react-core";
+import {
+  TableComposable,
+  Thead,
+  Tr,
+  Th,
+  Tbody,
+  Td,
+} from "@patternfly/react-table";
+import axios from "axios";
+import BarsIcon from "@patternfly/react-icons/dist/js/icons/bars-icon";
+import SearchBox from "../SearchComponent";
+import DatePickerWidget from "../DatePickerComponent";
+import Heading from "../HeadingComponent";
+import PathBreadCrumb from "../BreadCrumbComponent";
+import AlertMessage from "../AlertComponent";
+import NavItems from "../NavbarComponent";
+import EmptyTable from "../EmptyStateComponent";
+import moment from "moment";
+import { fetchPublicDatasets } from "../../../actions/fetchPublicDatasets";
+let startDate = moment(new Date(1990, 10, 4)).format("YYYY/MM/DD");
+let endDate = moment(new Date(2040, 10, 4)).format("YYYY/MM/DD");
+let controllerName = "";
+let dataArray = [];
+export const TableWithFavorite = () => {
+  const columnNames = {
+    controller: "Controller",
+    name: "Name",
+    creationDate: "Created On",
+  };
+  const [activeSortIndex, setActiveSortIndex] = useState(null);
+  const [activeSortDirection, setActiveSortDirection] = useState(null);
+  const [favoriteRepoNames, setFavoriteRepoNames] = useState([]);
```

Comment:
FYI: Nothing to be done about this right now; but we want to move towards allowing an authenticated user to manage per-user "metadata" on the server, so that things like "favorites" can be persisted across sessions. (We'll need to figure out exactly how that works.) Of course an unauthenticated user can only have the local browser Redux store because they can't write to the server...

Reply:
One related point to raise:  the theme for this PR is "non-logged in users" (which is fine), but non-logged in users are not going to be the central focus of the project -- we want it to work nicely for logged-in users and also work reasonably for non-logged in users.  And, we should minimize the amount of code and effort that we put in to supporting non-logged in users.  So, the code should target the logged-in case, and it should also handle the situation if the requestor is not logged in.  (We shouldn't have two completely independent sets of code for logged-in and non-logged-in users.)

Reply:
Yes,no extra set of code would be written to support authenticated users.To support the logged-in users,minor changes with respect to API and getting logged-in user info will be done in the current Table component only. 

Table component is a general component which will handle both non-logged-in and logged-in users.It is written in a way that it can support logged-in users also when we work on that part.